api/lagoon/v1beta2/lagoonbuild_helpers.go

```diff
@@ -277,6 +287,146 @@ func LagoonBuildPruner(ctx context.Context, cl client.Client, cns string, builds
 	}
 }
 
+// handle deleting any external resources here
+func DeleteBuildPod(
+	ctx context.Context,
+	cl client.Client,
+	opLog logr.Logger,
+	lagoonBuild *LagoonBuild,
+	nsType types.NamespacedName,
+	cns string,
+) error {
+	// get any running pods that this build may have already created
+	lagoonBuildPod := corev1.Pod{}
+	err := cl.Get(ctx, types.NamespacedName{
+		Namespace: lagoonBuild.ObjectMeta.Namespace,
+		Name:      lagoonBuild.ObjectMeta.Name,
+	}, &lagoonBuildPod)
+	if err != nil {
+		// handle updating lagoon for a deleted build with no running pod
+		// only do it if the build status is Pending or Running though
+		return fmt.Errorf("unable to find a build pod for %s", lagoonBuild.ObjectMeta.Name)
+	} else {
```

Comment:
You could simplify by dropping this "else" I think, that would get rid of the extraneous return at the end of the function